src/backend/distributed/executor/subplan_execution.c

```diff
@@ -36,7 +36,10 @@ ExecuteSubPlans(DistributedPlan *distributedPlan)
 	List *subPlanList = distributedPlan->subPlanList;
 	ListCell *subPlanCell = NULL;
 	List *nodeList = NIL;
-	bool writeLocalFile = false;
+	ListCell *nodeCell = NULL;
+	bool writeLocalFile = true;
+	int32 localGroupId = GetLocalGroupId();
```

Comment:
[4:12 PM] Marco Slot
    in follower clusters the nodes will have the same group ID since the pg_dist_local_group replicates
​[4:12 PM] Marco Slot
    so you should also look at the cluster name
​[4:13 PM] Marco Slot
    I think it should be possible to write a GetLocalNodeId() function that looks for the current group ID and cluster name in pg_dist_node

Reply:
Actually, we only need to check `writeToLocalFile = GetLocalGroupId() == 0`, because if I'm a worker I will always write to myself (until pruning is implemented)